src/GovUk.Education.ExploreEducationStatistics.Content.Model/Repository/ReleaseRepository.cs

```diff
@@ -18,6 +18,39 @@ public ReleaseRepository(ContentDbContext contentDbContext)
         _contentDbContext = contentDbContext;
     }
 
+    public async Task<DateTime> GetPublishedDate(Guid releaseId,
+        DateTime actualPublishedDate)
+    {
+        var release = await _contentDbContext.Releases
+            .SingleAsync(r => r.Id == releaseId);
+
+        // If the release is already published return the published date
+        if (release.Published.HasValue)
+        {
+            return release.Published.Value;
+        }
```

Comment:
I wonder if the overhead of making this release fetch async is worthwhile.

I wonder what's going on under the hood here. I think it'll create another process to get the release, and then immediately pass that information back to the initial thread? My intuition is that that is more costly than than just doing it all on the same thread.

Not saying change this - just thinking out loud :)

Reply:
I think it's probably a tiny difference especially as it should be called so infrequently, either when publishing a release or lazily caching a release. Maybe we should discuss this at the dev's team meeting to get a better understanding of when to use async and the benefits of it.

I had thought that anything that's potentially going to be waiting for the database is better to use async in case there's any delay it, allowing the calling method to continue execution in the mean time.